ui/src/admin/actions/index.js

```diff
@@ -1,18 +1,50 @@
 import {
-  getUsers,
-  getRoles,
+  getUsers as getUsersAJAX,
+  getRoles as getRolesAJAX,
+  createUser as createUserAJAX,
   deleteRole as deleteRoleAJAX,
   deleteUser as deleteUserAJAX,
 } from 'src/admin/apis'
 import {killQuery as killQueryProxy} from 'shared/apis/metaQuery'
 
+import {publishNotification} from 'src/shared/actions/notifications';
+
+import {ADMIN_NOTIFICATION_DELAY} from 'shared/constants'
+
 export const loadUsers = ({users}) => ({
   type: 'LOAD_USERS',
   payload: {
     users,
   },
 })
 
+export const loadRoles = ({roles}) => ({
+  type: 'LOAD_ROLES',
+  payload: {
+    roles,
+  },
+})
+
+export const addUser = () => ({
+  type: 'ADD_USER',
+})
+
+export const confirmUserCreated = (user, createdUser) => ({
+  type: 'CONFIRM_USER_CREATED',
```

Comment:
A bit nit picky but would this be better named 'USER_CREATE_SUCCESS' .  Correspondingly, there could be a 'USER_CREATE_ERROR' that would undo the optimistic update?